afew/MailMover.py

```diff
@@ -72,31 +75,45 @@ def move(self, maildir, rules):
                     if self.dry_run:
                         continue
                     try:
-                        shutil.copy2(fname, self.get_new_name(fname, destination))
-                        to_delete_fnames.append(fname)
-                    except shutil.SameFileError:
-                        logging.warn("trying to move '{}' onto itself".format(fname))
-                        continue
+                        logging.info("Copy original mail: "+fname)
+                        logging.info("Path of mail: "+os.path.dirname(fname))
+                        logging.info("Copy to: "+destination)
+                        dbpath = self.db.get_path()
+                        logging.info("Path of DB: "+dbpath)
+                        relpath=os.path.relpath(os.path.dirname(destination), dbpath)
+                        filename = os.path.basename(fname)
+                        logging.info("filename: "+filename)
+                        logging.info("Relative path DB -> file: " + relpath)
+                        # 1 COPY
+                        shutil.copy2(fname, self.get_new_name(
+                                                            fname,
+                                                            destination))
+                        dest_file_rel = dbpath+'/'+relpath+'/'+filename
+                        logging.info("Add new copy to the DB: "+dest_file_rel)
+                        # OPEN DB R/W
+                        logging.info("Open DB R/W")
+                        # print vars(self.db)
+                        # 2 Add NEW MESSAGE TO DB
+                        moved_message = notmuch.Database.add_message(self.db, dest_file_rel)
+                        logging.info("Delete original mail file: "+fname)
+                        os.remove(fname)
+                        # 3 REMOVE ORIGINAL
+                        logging.info("Delete the original mail in DB")
+                        # 4 REMOVE ORIGINAL FROM DB
+                        notmuch.Database.remove_message(self.db, fname)
+                    except IOError as e:
+                        logging.warn('Error IO: '+format(e))
                     except shutil.Error as e:
-                        # this is ugly, but shutil does not provide more
-                        # finely individuated errors
-                        if str(e).endswith("already exists"):
-                            continue
```

Comment:
What's with this line? Do we already check somewhere else if source and destination are the same, which is why we skip that check?

Reply:
I removed that because this will be logged by 'shutil' log and I think if you try to copy over the same thing it should be an error from shutil and/or your FS but I will add a test with a similar message for that case.